query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java

```diff
@@ -450,7 +450,9 @@ private String convertLiteralToString(LiteralConstant literal, Params.Builder pa
     String ret = null;
     switch (value.getValueType()) {
       case STRING_ARRAY:
-        ret = buildArrayValue(value.getStringArrayList(), paramsBuilder::addStringParam);
+        List<String> values =
+            value.getStringArrayList().size() > 0 ? value.getStringArrayList() : List.of("null");
```

Comment:
why are we handling special case only for string array? There are other array types as well. We don't need to handle this case for them?